solidity/contracts/bridge/Bridge.sol

```diff
@@ -1221,88 +1270,37 @@ contract Bridge is Ownable, EcdsaWalletOwner {
             bytes memory outputScript = output.slice(8, output.length - 8);
 
             if (
-                info.changeValue == 0 &&
-                (keccak256(outputScript) == walletP2PKHScriptKeccak ||
-                    keccak256(outputScript) == walletP2WPKHScriptKeccak) &&
+                resultInfo.changeValue == 0 &&
+                (keccak256(outputScript) ==
+                    processInfo.walletP2PKHScriptKeccak ||
+                    keccak256(outputScript) ==
+                    processInfo.walletP2WPKHScriptKeccak) &&
                 outputValue > 0
             ) {
                 // If we entered here, that means the change output with a
                 // proper non-zero value was found.
-                info.changeIndex = uint32(i);
-                info.changeValue = outputValue;
+                resultInfo.changeIndex = uint32(i);
+                resultInfo.changeValue = outputValue;
             } else {
                 // If we entered here, that the means the given output is
                 // supposed to represent a redemption. Build the redemption key
                 // to perform that check.
```

Comment:
Redemption key is no longer built here but within `processNonChangeRedemptionTxOutput`.

Reply:
027b0bab7715569acf2430189f0b3bf1c14ec3d4